### PR TITLE
[Feature] Create SemanticKITTI pkl

### PR DESCRIPTION
**Pull request by xizaoqu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Create SemanticKITTI pkl

## Modification

Add SemanticKITTI pkl creation in create_data.py

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: tools/dataset_converters/semantickitti_converter.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+}
+split_list = ['train', 'valid', 'test']
+
+
+def get_semantickitti_info(split):
+    data_infos = dict()
+    data_infos['metainfo'] = dict(DATASET='SemanticKITTI')
+    data_list = []
+    for i_folder in fold_split[split]:
+        for j in range(0, total_num[i_folder]):
```
Comment (Tai-Wang) [inline tools/dataset_converters/semantickitti_converter.py:43]: may add a comment to provide an example filename of lidar_points, pts_semantic_mask_path and sample_id

File: tools/dataset_converters/semantickitti_converter.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    return data_infos
+
+
+def create_semantickitti_info_file(pkl_prefix='semantickitti', save_path=None):
+    """Create info file of SemanticKITTI dataset.
+
+    Directly generate info file without raw data.
+
+    Args:
+        pkl_prefix (str, optional): Prefix of the info file to be generated.
```
Comment (Tai-Wang) [inline tools/dataset_converters/semantickitti_converter.py:68]: no need for "optional" (now we only use optional for the variable that is set to None by default)

File: tools/dataset_converters/semantickitti_converter.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    filename = save_path / f'{pkl_prefix}_infos_val.pkl'
+    print(f'SemanticKITTI info val file is saved to {filename}')
+    mmengine.dump(semantickitti_infos_val, filename)
+    semantickitti_infos_test = get_semantickitti_info(split='test')
+    filename = save_path / f'{pkl_prefix}_infos_test.pkl'
+    print(f'SemanticKITTI info test file is saved to {filename}')
+    mmengine.dump(semantickitti_infos_test, filename)
+
+
+create_semantickitti_info_file('semantickitti', 'data/semantickitti')
```
Comment (ZwwWayne) [inline tools/dataset_converters/semantickitti_converter.py:104]: we usually implement a main function and use arguments, see other files for example.

Comment (xizaoqu) [reply tools/dataset_converters/semantickitti_converter.py]: Done. I forgot to remove this line.

**Review by Tai-Wang — APPROVED** [review]

**Pull request merged**